plugins/rest/aws.go

```diff
@@ -318,6 +318,150 @@ func (cs *awsMetadataCredentialService) credentials(ctx context.Context) (aws.Cr
 	return cs.creds, nil
 }
 
+// awsCustomIdentityCredentialService represents an STS credential service that uses active IAM credentials
+// to obtain temporary security credentials generated by AWS STS for allowing existing IAM users access to AWS resources
+// that they don't already have access to.
+type awsCustomIdentityCredentialService struct {
+	RoleArn          string
+	AccessKey        string
+	SecretKey        string
+	SessionToken     string
+	RegionName       string `json:"aws_region"`
+	SessionName      string `json:"session_name"`
+	Domain           string `json:"aws_domain"`
+	stsURL           string
+	creds            aws.Credentials
+	expiration       time.Time
+	logger           logging.Logger
+	signatureVersion string
+}
+
+func (cs *awsCustomIdentityCredentialService) populateFromEnv() error {
+	cs.RoleArn = os.Getenv(awsRoleArnEnvVar)
+	if cs.RoleArn == "" {
+		return errors.New("no " + awsRoleArnEnvVar + " set in environment")
+	}
+
+	cs.AccessKey = os.Getenv(accessKeyEnvVar)
+	if cs.AccessKey == "" {
+		return errors.New("no " + accessKeyEnvVar + " set in environment")
+	}
+	cs.SecretKey = os.Getenv(secretKeyEnvVar)
+	if cs.SecretKey == "" {
+		return errors.New("no " + secretKeyEnvVar + " set in environment")
+	}
```

Comment:
@ashutosh-narkar will OPA not be able to obtain these credentials from EC2 metadata service using aws.GetDefaultConfig() if they have not been provided in the environment?

Reply:
This will require OPA consumer to populate these credentials and keep them up to date.

Reply:
Added support for credentials fetching using Environment variables, Credentials file and EC2 metadata service.